mlflow/models/utils.py

```diff
@@ -547,7 +562,11 @@ def _enforce_mlflow_datatype(name, values: pd.Series, t: DataType):
     # NB: Comparison of pandas and numpy data type fails when numpy data type is on the left hand
     # side of the comparison operator. It works, however, if pandas type is on the left hand side.
     # That is because pandas is aware of numpy.
-    if t.to_pandas() == values.dtype or t.to_numpy() == values.dtype:
+    if (
+        t.to_pandas() == values.dtype
+        or t.to_numpy() == values.dtype
+        or t.to_python() == values.dtype
```

Comment:
The type of values will still be be pd.series because it is converted into pandas. I added this line as it is unable to compare Integer and int properly.